File: adapter-pattern/python/src/Mp4Player.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

from src.MediaPlayer import MediaPlayer


class Mp4Player(MediaPlayer):

    def play(self, ):
        print('Mp4Player file_type:' + self.file_type + ' | file_path:' +
              self.file_path)
        print('Mp4Player::play()')

File: proxy-pattern/java/src/Mp4Player.java
package src;

// HlsPlayer播放器实现了视频接口
public class Mp4Player implements VideoPlayer {
  String fileType = null;
  String filePath = null;

  public Mp4Player(String fileType, String filePath) {
    this.fileType = fileType;
    this.filePath = filePath;
  }

  public void play() {
    System.out.println("Mp4Player path:" + filePath);
    System.out.println("Mp4Player::play()");
  }
}
